features/FEATURE_BLE/targets/TARGET_NORDIC/TARGET_NORDIC_CORDIO/TARGET_NRF5x/NRFCordioHCIDriver.cpp

```diff
@@ -361,9 +361,16 @@ void NRFCordioHCIDriver::start_reset_sequence()
 
 bool NRFCordioHCIDriver::get_random_static_address(ble::address_t& address)
 {
-    PalCfgLoadData(PAL_CFG_ID_BD_ADDR, address.data(), sizeof(bdAddr_t));
-
-    MBED_ASSERT((address[5] & 0xC0) == 0xC0);
+    /* address cannot have all bits set to 1 */
+    bool no_binary_zeros = true;
```

Comment:
Wouldn't it be better to get the address from `NRF_FICR->DEVICEID` so we always get the same random static address ?

Reply:
why not, I'll change it